packages/labs/task/src/task.ts

```diff
@@ -102,71 +113,109 @@ export class Task<T extends [...unknown[]] = any, R = any> {
    * is kept and only resolved when the new run is completed.
    */
   taskComplete!: Promise<R>;
+
+  /**
+   * Controls if they task will run when its arguments change. Defaults to true.
+   */
+  autoRun = true;
+
   private _resolveTaskComplete!: (value: R) => void;
   private _rejectTaskComplete!: (e: unknown) => void;
 
   constructor(
     host: ReactiveControllerHost,
-    task: TaskFunction<T, R>,
-    getDependencies: DepsFunction<T>
+    task: TaskFunction<T, R> | TaskConfig<T, R>,
```

Comment:
Let's write this as two overloads: one with task and getArgs and the other with just a config.

Reply:
Done.